### ISSUE TITLE
Android - Distance - A deleted offline waypoint remains in the distance editor after going online

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#40589]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.63-7
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4498589
**Email or phone of affected tester (no customers):** sustinov@applausemail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Prerequisites:
Create WS collect, invite an employee.

Step:
1. Open https://staging.new.expensify.com/
2. Log in to the employee account
3. Navigate to a workspace room
4. Create a 2 IOU distance with 3 waypoints
5. Navigate to the last IOU you created
6. Switch off the Internet
7. Change the date and Save
8. Navigate to the waypoint editing menu
9. Delete the last waypoint
10. Save
11. Switch on the internet
12. Return to the Report Conversation
13. Once the IOU is fully registered, go to the waypoint editing menu

## Expected Result:
A deleted waypoint in offline should disappear after going online

## Actual Result:
A deleted offline waypoint remains in the distance editor after going online

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/da4b8ff1-e389-4913-a1a8-418acd4d2885

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01cbc67100f420bf95</li>
        <li>Upwork Job ID: 1785502891993055232</li>
        <li>Last Price Increase: 2024-06-02</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 102965910</li>
<li>paultsimura | Contributor | 102965911</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by izarutskaya** [comment]
We think this issue might be related to the #collect project.

**Comment by mallenexpensify** [comment]
Checking on [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1713912840962429)
 
I have this issue, which is for Android Native, couple questions
Didn't Browserstack used to automatically load the NewDot app installed each time it loaded? 
Are there known issues with Android deploys now? If so, do we want to take any actions based on that? ie. put on hold.

**Comment by mallenexpensify** [comment]
This is distance related so adding to #wave-collect. Unable to repro cuz app didn't auto-load via browserstack.

**Comment by mallenexpensify** [comment]
@shubham1206agra , do you have an Android phone and are you able to repro?

**Comment by mallenexpensify** [comment]
Thanks @shubham1206agra , I just tried to repro via Browserstack and got stuck on the loading screen. Seems like this can be worked on externally, added the label.

**Comment by mallenexpensify** [comment]
Waiting on proposals

**Comment by mallenexpensify** [comment]
@shubham1206agra , yeah, bumped to $500.  Thx for the ping.

**Comment by mallenexpensify** [comment]
Waiting on proposals

**Comment by mallenexpensify** [comment]
@shubham1206agra , 👀 on @dominictb 's (in-depth) proposal above plz

**Comment by mallenexpensify** [comment]
Few levels deep here to get to the PR, posted there, bumped to weekly and put on hold (removed help wanted)
https://github.com/Expensify/react-native-onyx/pull/490#issuecomment-2145716489

**Comment by mallenexpensify** [comment]
@paultsimura please comment so I can assign, added to
- https://github.com/Expensify/App/issues/44582

**Comment by mallenexpensify** [comment]
Added the label, it might take a bit.  I don't have a physical android phone so I can't quickly/easily test.

**Comment by mallenexpensify** [comment]
Bumped to daily, made me the owner and requested a retest in the QA channel
https://expensify.slack.com/archives/C9YU7BX5M/p1725410212524379

**Comment by mvtglobally** [comment]
Issue is not reproducible https://github.com/user-attachments/assets/be89fbad-34e9-45c8-b86c-aa15324d89b3

**Comment by mallenexpensify** [comment]
Thanks @mvtglobally , closing
